File: samples/Sample.rb
# frozen_string_literal: true

require "json"
require "net/http"

MAX_RETRIES = 3
VERSION = "1.0.0"

module IslandsDark
  class Color
   attr_reader :r, :g, :b

    def initialize(r, g, b)
      @r = r
      @g = g
      @b = b
    end

    def to_hex
      format("#%02X%02X%02X", @r, @g, @b)
    end

    def self.from_hex(hex)
      hex = hex.delete("#")
      Color.new(
        hex[0..1].to_i(16),
        hex[2..3].to_i(16),
        hex[4..5].to_i(16)
      )
    end
  end

  class Theme
    attr_reader :name, :colors

    def initialize(name)
      @name = name
      @colors = {}
    end

    def add_color(key, color)
      @colors[key] = color
    end

    def dark?
      true
    end

    def to_json(*_args)
      {
        name: @name,
        dark: true,
        colors: @colors.transform_values(&:to_hex)
      }.to_json
    end
  end

  module_function

  def process_items(items, prefix: "item")
    items.each_with_index.map { |item, i| "#{prefix}_#{i}: #{item}" }
  end
end

theme = IslandsDark::Theme.new("Islands Dark")
theme.add_color("keyword", IslandsDark::Color.from_hex("#CF8E6D"))
theme.add_color("background", IslandsDark::Color.new(25, 26, 28))

items = IslandsDark.process_items(["alpha", "beta", "gamma"])

=begin
This is a multi-line
block comment in Ruby
=end

puts "Theme: #{theme.name}"
items.each { |item| puts "  - #{item}" }

File: samples/Sample.php
<?php

declare(strict_types=1);

namespace IslandsDark\Sample;

const MAX_RETRIES = 3;
const VERSION = '1.0.0';
const PI = 3.14159;

class Color
{
    private int $r;
    private int $g;
    private int $b;

    public function __construct(int $r, int $g, int $b)
    {
        $this->r = $r;
        $this->g = $g;
        $this->b = $b;
    }

    public function toHex(): string
    {
        return sprintf('#%02X%02X%02X', $this->r, $this->g, $this->b);
    }

    public static function fromHex(string $hex): self
    {
        $hex = ltrim($hex, '#');
        return new self(
            hexdec(substr($hex, 0, 2)),
            hexdec(substr($hex, 2, 2)),
            hexdec(substr($hex, 4, 2))
        );
    }
}

#[\Attribute]
class ThemeAttribute
{
    public function __construct(public string $name, public bool $dark = true) {}
}

#[ThemeAttribute(name: 'Islands Dark')]
class Theme
{
    private array $colors = [];
    private string $name;

    public function __construct(string $name)
    {
        $this->name = $name;
    }

    public function addColor(string $key, Color $color): void
    {
        $this->colors[$key] = $color;
    }

    public function getColor(string $key): ?Color
    {
        return $this->colors[$key] ?? null;
    }

    public function getName(): string
    {
        return $this->name;
    }
}

function processItems(array $items, string $prefix = 'item'): array
{
    $result = [];
    foreach ($items as $i => $item) {
        $result[] = "{$prefix}_{$i}: {$item}";
    }
    return $result;
}

$theme = new Theme('Islands Dark');
$theme->addColor('keyword', Color::fromHex('#CF8E6D'));
$items = processItems(['alpha', 'beta', 'gamma']);

echo "Theme: " . $theme->getName() . "\n";

?>